src/normalizer.ts

```diff
@@ -338,30 +339,27 @@ class V1SchemaNormalizer {
     if (!s || typeof s !== 'object' || Array.isArray(s)) return undefined
     if (s.xtpType) return s.xtpType
 
-    if ((s.type && s.type === 'object') ||
-      (s.properties && s.properties.length > 0)) {
-
+    if (s.properties && s.properties.length > 0) {
       s.type = 'object'
 
       const properties: XtpNormalizedType[] = []
-      if (s.properties) {
-        for (const pname in s.properties) {
-          const p = s.properties[pname]
-
-          const t = this.annotateType(p, [...path, 'properties', p.name ?? pname])
-          if (t) {
-            p.xtpType = t
-            properties.push(t)
-          }
+      for (const pname in s.properties) {
+        const p = s.properties[pname]
+
+        const t = this.annotateType(p, [...path, 'properties', p.name ?? pname])
+        if (t) {
+          p.xtpType = t
+          properties.push(t)
         }
-        return new ObjectType(s.name || '', properties, s)
-      } else {
-        // untyped object
-        return new ObjectType('', [], s)
       }
+      return new ObjectType(s.name || '', properties, s)
```

Comment:
A name is required for ObjectTypes right? maybe we should change this to `return new ObjectType(s.name!, properties, s)`

Reply:
It should be, fixed in e59ec71c.